File: Weekly294/Q2_maximumBags.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        
        remain = []
        res = 0
        
        for i in range(len(capacity)):
            remain.append(capacity[i]-rocks[i])
            
        remain.sort()
        
        for i in range(len(remain)):
            if remain[i] == 0:
                res+=1
            elif additionalRocks >= remain[i]:
                additionalRocks -= remain[i]
                res += 1
            elif additionalRocks < remain[i]:
                break
        return res

File: Weekly294/Q2_maximumBags.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        vector<int> c;
        
        int atFull = 0;
        int diff;
        
        for (int i = 0; i < capacity.size(); i++) {
            diff = capacity[i] - rocks[i];
            if (diff == 0) {
                ++atFull;
                continue;
            }
            
            c.push_back(diff);
        }
        
        sort(c.begin(), c.end());
        
        for(int i = 0; i < c.size(); i++) {
            additionalRocks -= c[i];
            if (additionalRocks < 0) {
                return atFull;
            } else if (additionalRocks > 0) {
                ++atFull;
                continue;
            }
            ++atFull;
            break;
        }
        return atFull;
    }
};